torch/nested/_internal/ops.py

```diff
@@ -2172,16 +2189,40 @@ def to_padded_tensor_default(func, *args, **kwargs):
     return padded_out
 
 
-@register_jagged_func(
-    torch.ops.aten._nested_from_padded_tensor.default,
-    "padded: t, offsets: t, dummy: jt, ragged_idx: any?, min_seqlen: any?, max_seqlen: any?, sum_S: any?",
-)
+from torch.nested._internal.cached_tensor import register_cached_tensor_func
+
+
+# When we use [ CacheTensor open registry ], we cannot rely on NestedTensor's
+# torch function anymore to enable maybe_enable_thunkify. All functions
+# registered to register_cached_tensor_func must be first wrapped with this
+# decorator.
```

Comment:
I'm not exactly sure what's going on here, but can't you combine `register_cached_tensor_func()` + `maybe_enable_thunkify()` into a single decorator to avoid this weird dependency on wrapped functions?

Reply:
Do you mean to add a helper specific to nested tensor? And the concern that there'd be less room for error for anyone adding a new constructor? 

I guess some counter points are that (1) maybe_enable_thunkify is temporary anyway, if I do the combined approach, once maybe_enable_thunkify is removed, then I'd have to refactor back to remove the helper which is no obsolete. (2) I don't imagine anyone adding new constructors very often.

If you feel strongly about this, happy to update though.

Reply:
> (1) maybe_enable_thunkify is temporary anyway, if I do the combined approach, once maybe_enable_thunkify is removed, then I'd have to refactor back to remove the helper which is no obsolete

the helper could theoretically stay in place though and its impl could change to no longer use maybe_enable_thunkify underneath. So from that perspective, it's an abstraction layer that shields you from the future change.

That said, I don't have too strong opinions here. I think your points that this doesn't come up much help. Does the ordering of the decorators matter?

Reply:
> Does the ordering of the decorators matter?
 
 Yes, you need to register the decorated function